File: src/types/Property.ts
export interface Property {
  id?: string;
  idOwner: string;
  name: string;
  address: string;
  price: number;
  image: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreatePropertyDto {
  idOwner: string;
  name: string;
  address: string;
  price: number;
  image: string;
}

export interface UpdatePropertyDto {
  id?: string;
  idOwner: string;
  name: string;
  address: string;
  price: number;
  image: string;
}

export interface PropertyFilterDto {
  name?: string;
  address?: string;
  minPrice?: number;
  maxPrice?: number;
  idOwner?: string;
  page?: number;
  pageSize?: number;
}

export interface PropertyResponse {
  properties: Property[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

File: backend/RealEstate.API/Models/Property.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace RealEstate.API.Models
{
    public class Property
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonElement("idOwner")]
        public string IdOwner { get; set; } = string.Empty;

        [BsonElement("name")]
        public string Name { get; set; } = string.Empty;

        [BsonElement("address")]
        public string Address { get; set; } = string.Empty;

        [BsonElement("price")]
        public decimal Price { get; set; }

        [BsonElement("image")]
        public string Image { get; set; } = string.Empty;

        [BsonElement("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [BsonElement("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
